sfn/src/main/scala/uk/gov/nationalarchives/DASFNClient.scala

```diff
@@ -0,0 +1,61 @@
+package uk.gov.nationalarchives
+
+import cats.effect.Async
+import io.circe.{Encoder, Printer}
+import io.circe.syntax._
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sfn.SfnAsyncClient
+import software.amazon.awssdk.services.sfn.model.{StartExecutionRequest, StartExecutionResponse}
+
+/** An SFN client. It is written generically so can be used for any effect which has an Async instance. Requires an
+  * implicit instance of cats Async which is used to convert CompletableFuture to F
+  *
+  * @param sfnAsyncClient
+  *   An AWS SFN Async client
+  * @tparam F
+  *   Type of the effect
+  */
+class DASFNClient[F[_]: Async](sfnAsyncClient: SfnAsyncClient) {
+
+  /** @param stateMachineArn
+    *   The arn of the state machine to start
+    * @param input
+    *   A case class. This will be deserialised to a json string and sent as input to the step function.
+    * @param name
+    *   An optional step function name. If this is omitted, AWS will generate a UUID for a name.
+    * @param enc
+    *   A circe encoder which will encode the case class to JSON
+    * @tparam T
+    *   The type of the input case class
+    * @return
+    *   The response from the startExecution call, wrapped in F[_]
+    */
+  def startExecution[T <: Product](stateMachineArn: String, input: T, name: Option[String] = None)(implicit
```

Comment:
I wrote this comment but forgot to send it. Do you think it's worth adding a trait called something like `Input` with a property called "Task" (as all input jsons seem to have this key in common) and then when it comes to creating your input, the user would need to mix in the trait with the case class?

Reply:
So we'd have something like 
```
trait Input {
// What do we do here?
}
```
And then in the method we can do 
`def startExecution[T <: Input](stateMachineArn: String, input: T`

Then if the calling client wants to send something, they have to do something like
```
MyCallingCaseClass(stuff: String, to: String, send: String) extends Input
```

Reply:
Yes. I was thinking something like:

```
  trait Input {
    val Task: String
  }
```

Reply:
🤦 Sorry, I meant to write "Type" as I noticed we usually have this in the step function input

```
  "The Task": {
      "Type": "Choice",
      ...
    }
```

So I was thinking of adding it to help whoever is using the sfn client.